### DISCUSSION TITLE
[`webpack-dev-server`] On build hook

### DISCUSSION THREAD

**Discussion by Yash-Singh1** [webpack/webpack#15982] [General]
## Feature request

**What is the expected behavior?**

There is a hook in the dev server library that lets you listen for a rebuild of the app.

**What is motivation or use case for adding/changing the behavior?**

Sometimes you want to recompile and add in static assets manually in wrappers of the webpack dev server and a hook where you can specify a function to do that makes it easier.

**How should this be implemented in your opinion?**

Simple function in the configuration that is called every time a watch rebuild is triggered.

**Are you willing to work on this yourself?**
no

**Comment by alexander-akait** [comment]
We have the `done` hook - https://webpack.js.org/api/compiler-hooks/#done, it called every time when compiler finished compilation(compilations if you have multi compiler mode)

**Reply by Yash-Singh1** [reply]
Is there any way I can specify this to the `webpack-dev-server` options?
https://webpack.js.org/configuration/dev-server/#devserver

**Reply by alexander-akait** [reply]
You should not specify it for dev server, it is a hook for compiler, so you can create simple plugin and use done hook to run your code

**Reply by Yash-Singh1** [reply]
My usecase was where I had to specify it to the `webpack-dev-server`, and I was looking for an option on that similar to `onListening` and `onAfterSetupMiddleware`

**Reply by alexander-akait** [reply]
Why don't use hook? dev server starts and emit compilations using hooks
